### ISSUE TITLE
[ONNX] torch.repeat_interleave export failed

### ISSUE THREAD

**Issue by ilyasher** [pytorch/pytorch#100429]
### 🐛 Describe the bug

I encountered an error when exporting a model with a repeat_interleave op. Here is a minimal repro:

```python
import torch
import torch.nn as nn

class MyModel(nn.Module):
    def forward(self, x):
        return x.repeat_interleave(2)

model = MyModel()
args = (torch.ones(1, 3),)

print("model.forward() succeeeds: ", model(*args).shape)

torch.onnx.export(
    model,
    args,
    "repeat_interleave.onnx",
    input_names=["input"], 
    output_names=["output"], 
    dynamic_axes={"input": [0]},
    opset_version=17,
)
```

It seems to happen when there are dynamic shapes and the `dim` keyword is not specified in `torch.repeat_interleave`.

Here is the error:
```
model.forward() succeeeds:  torch.Size([6])
/home/isherstyuk/repos/pytorch/torch/onnx/utils.py:2050: UserWarning: No names were found for specified dynamic axes of provided input.Automatically generated names will be applied to each dynamic axes of input input
  warnings.warn(
========= Diagnostic Run torch.onnx.export version 2.1.0a0+git0cf6e74 ==========
verbose: False, log level: Level.ERROR
======================= 0 NONE 0 NOTE 0 WARNING 0 ERROR ========================

Traceback (most recent call last):
  File "/home/isherstyuk/repos/pytorch/pyt-test.py", line 13, in <module>
    torch.onnx.export(
  File "<@beartype(torch.onnx.utils.export) at 0x7ffb7d75ad30>", line 363, in export
  File "/home/isherstyuk/repos/pytorch/torch/onnx/utils.py", line 507, in export
    _export(
  File "/home/isherstyuk/repos/pytorch/torch/onnx/utils.py", line 1567, in _export
    graph, params_dict, torch_out = _model_to_graph(
  File "<@beartype(torch.onnx.utils._model_to_graph) at 0x7ffb7d6e5b80>", line 11, in _model_to_graph
  File "/home/isherstyuk/repos/pytorch/torch/onnx/utils.py", line 1128, in _model_to_graph
    graph = _optimize_graph(
  File "<@beartype(torch.onnx.utils._optimize_graph) at 0x7ffb7d6e5040>", line 90, in _optimize_graph
  File "/home/isherstyuk/repos/pytorch/torch/onnx/utils.py", line 666, in _optimize_graph
    graph = _C._jit_pass_onnx(graph, operator_export_type)
  File "<@beartype(torch.onnx.utils._run_symbolic_function) at 0x7ffb7d6e8c10>", line 97, in _run_symbolic_function
  File "/home/isherstyuk/repos/pytorch/torch/onnx/utils.py", line 1908, in _run_symbolic_function
    return symbolic_fn(graph_context, *inputs, **attrs)
  File "<@beartype(torch.onnx.symbolic_opset13.repeat_interleave) at 0x7ffb7d6c8160>", line 32, in repeat_interleave
  File "/home/isherstyuk/repos/pytorch/torch/onnx/symbolic_opset13.py", line 645, in repeat_interleave
    reps = symbolic_helper._size_helper(g, input, dim)
  File "<@beartype(torch.onnx.symbolic_helper._size_helper) at 0x7ffb7d90cd30>", line 32, in _size_helper
  File "/home/isherstyuk/repos/pytorch/torch/onnx/symbolic_helper.py", line 1322, in _size_helper
    return select(g, full_shape, g.op("Constant", value_t=torch.tensor([0])), dim)
  File "/home/isherstyuk/repos/pytorch/torch/onnx/symbolic_helper.py", line 392, in wrapper
    return fn(g, *args, **kwargs)
  File "/home/isherstyuk/repos/pytorch/torch/onnx/symbolic_helper.py", line 306, in wrapper
    return fn(g, *args, **kwargs)
  File "<@beartype(torch.onnx.symbolic_opset9.select) at 0x7ffb7d8904c0>", line 32, in select
  File "/home/isherstyuk/repos/pytorch/torch/onnx/symbolic_opset9.py", line 1210, in select
    return g.op("Gather", self, index, axis_i=dim)
  File "<@beartype(torch.onnx._internal.jit_utils.GraphContext.op) at 0x7ffb7d95dca0>", line 38, in op
beartype.roar.BeartypeCallHintParamViolation: @beartyped torch.onnx._internal.jit_utils.GraphContext.op() parameter raw_args=0 violates type hint typing.Union[torch.Tensor, torch.Value], as int 0 not <class "torch.Value"> or <protocol "torch.Tensor">.
```

Note: the export succeeds when I provide the `dim` keyword to `torch.repeat_interleave`, but that changes the behavior. A workaround is to replace `x.repeat_interleave(2)` with `x.flatten().repeat_interleave(2, dim=0)`, but in a perfect world it wouldn't be necessary.

[Docs for torch.repeat_interleave](https://pytorch.org/docs/stable/generated/torch.repeat_interleave.html)

Thanks!

### Versions

Versions of relevant libraries:
[pip3] flake8==6.0.0
[pip3] flake8-bugbear==23.3.23
[pip3] flake8-comprehensions==3.11.1
[pip3] flake8-executable==2.1.3
[pip3] flake8-logging-format==0.9.0
[pip3] flake8-pyi==23.3.1
[pip3] mypy==0.960
[pip3] mypy-extensions==1.0.0
[pip3] numpy==1.23.1
[pip3] torch==2.1.0a0+git0cf6e74
[pip3] torchvision==0.16.0.dev20230330+cpu
[conda] magma-cuda118             2.6.1                         1    pytorch
[conda] mkl                       2022.1.0           hc2b9512_224  
[conda] mkl-include               2022.1.0           h06a4308_224  
[conda] numpy                     1.23.1                   pypi_0    pypi
[conda] torch                     2.1.0a0+git0cf6e74           dev_0    <develop>
[conda] torchvision               0.16.0.dev20230330+cpu          pypi_0    pypi

**Comment by TaiPhamD** [comment]
I don't think this is a bug. The ONNX export syntax declared the input with dynamic size in the 0-th dimension:

```
dynamic_axes={"input": [0]},
```

So when you do **x.repeat_interleave(2)** the default behavior without specifying dimension is to [flatten your input tensor](https://github.com/pytorch/pytorch/blob/419387f66f39cb1f80c6ba1a01164b69fbb649fb/torch/onnx/symbolic_opset13.py#L606) before applying repeat_interleave.  But since the input has dynamic_axes then it makes it impossible for the [helper routine](https://github.com/pytorch/pytorch/blob/419387f66f39cb1f80c6ba1a01164b69fbb649fb/torch/onnx/symbolic_opset13.py#L617) to determine the input sizes correctly at export time since input size can only be computed during runtime with dynamic axis. 

Therefore , I think to support dynamic_axes you will have to change your model code to add dim=1 in the repeat_interleave (assuming your input dimension 0 is your batch size since you made it dynamic)

```
class MyModel(nn.Module):
    def forward(self, x):
        return x.repeat_interleave(2,dim=1)
```

**Comment by ilyasher** [comment]
@TaiPhamD `torch.onnx.export` raises an exception on a valid model, so this is a bug. The new flattened dimension has dynamic shape, which the export should still be able to handle. 

In any case, the fix is fairly simple and I'll include it in an upcoming PR. [This line](https://github.com/pytorch/pytorch/blob/419387f66f39cb1f80c6ba1a01164b69fbb649fb/torch/onnx/symbolic_opset13.py#L611) needs to be changed to initialize `dim` to a tensor instead of an int, to make the type consistent between the `if` and `else` clauses, and fixes the beartype error at the bottom of the stacktrace.

**Comment by ilyasher** [comment]
Fixed by #100575
